Name: Alia Ibrahim Bandarkar
PRN NO: 2030331245007

# import pandas and numpy libraries

In [2]:
import pandas as pd
import numpy as np

# 1. load a csv file

In [3]:
df=pd.read_csv('train.csv')


#displaying the contents of csv file

In [4]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

# 2. print the header of csv file

In [5]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

# 3. print the dimensional size of the dataset

In [6]:
df.shape

(891, 12)

# so our dataset contains 891 rows and 12 columns

# 4. print the name of variables or features

In [7]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

 # 5. print types of variables or features

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


#  6. print descriptive statistics of numeric variables

In [9]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

# 7. print descriptive statistics of non-numeric variables

In [10]:
df.describe(include='object')

Name   Sex Ticket        Cabin Embarked
count                               891   891    891          204      889
unique                              891     2    681          147        3
top     Goncalves, Mr. Manuel Estanslas  male   1601  C23 C25 C27        S
freq                                  1   577      7            4      644

# 8. print counts of distinct values for categorical variables

In [12]:
df['Sex'].value_counts()

male      577
female    314
Name: Sex, dtype: int64

In [13]:
df['Name'].value_counts()

Goncalves, Mr. Manuel Estanslas    1
Mallet, Master. Andre              1
Davies, Master. John Morgan Jr     1
Vander Cruyssen, Mr. Victor        1
Rice, Master. George Hugh          1
                                  ..
McGovern, Miss. Mary               1
Watson, Mr. Ennis Hastings         1
Ridsdale, Miss. Lucy               1
Chaffee, Mr. Herbert Fuller        1
Gustafsson, Mr. Johan Birger       1
Name: Name, Length: 891, dtype: int64

In [14]:
df['Ticket'].value_counts()

1601        7
CA. 2343    7
347082      7
CA 2144     6
347088      6
           ..
323951      1
364499      1
315151      1
PC 17612    1
250651      1
Name: Ticket, Length: 681, dtype: int64

In [15]:
df['Cabin'].value_counts()

C23 C25 C27    4
G6             4
B96 B98        4
D              3
F33            3
              ..
B19            1
B41            1
C85            1
D28            1
C70            1
Name: Cabin, Length: 147, dtype: int64

In [16]:
df['Embarked'].value_counts()

S    644
C    168
Q     77
Name: Embarked, dtype: int64

# 9. print percentage of distinct values for categorical variables

In [17]:
df['Sex'].value_counts(normalize="True")

male      0.647587
female    0.352413
Name: Sex, dtype: float64

In [18]:
df['Embarked'].value_counts(normalize="True")

S    0.724409
C    0.188976
Q    0.086614
Name: Embarked, dtype: float64

In [19]:
df['Cabin'].value_counts(normalize="True")

C23 C25 C27    0.019608
G6             0.019608
B96 B98        0.019608
D              0.014706
F33            0.014706
                 ...   
B19            0.004902
B41            0.004902
C85            0.004902
D28            0.004902
C70            0.004902
Name: Cabin, Length: 147, dtype: float64

In [20]:
df['Ticket'].value_counts(normalize="True")

1601        0.007856
CA. 2343    0.007856
347082      0.007856
CA 2144     0.006734
347088      0.006734
              ...   
323951      0.001122
364499      0.001122
315151      0.001122
PC 17612    0.001122
250651      0.001122
Name: Ticket, Length: 681, dtype: float64

In [21]:
df['Name'].value_counts(normalize="True")

Goncalves, Mr. Manuel Estanslas    0.001122
Mallet, Master. Andre              0.001122
Davies, Master. John Morgan Jr     0.001122
Vander Cruyssen, Mr. Victor        0.001122
Rice, Master. George Hugh          0.001122
                                     ...   
McGovern, Miss. Mary               0.001122
Watson, Mr. Ennis Hastings         0.001122
Ridsdale, Miss. Lucy               0.001122
Chaffee, Mr. Herbert Fuller        0.001122
Gustafsson, Mr. Johan Birger       0.001122
Name: Name, Length: 891, dtype: float64